common/logisticspipes/network/packets/SetCurrentPagePacket.java

```diff
@@ -46,11 +36,16 @@ public void processPacket(EntityPlayer player) {
 		ItemStack book;
 		book = player.getHeldItem(hand);
 		if (book.getItem() != LPItems.itemGuideBook) return;
-		NBTTagCompound nbt = book.getTagCompound();
-		if(nbt == null) nbt = new NBTTagCompound();
+		NBTTagCompound nbt;
+		if(book.hasTagCompound()){
+			nbt = book.getTagCompound();
+		}else{
+			nbt = new NBTTagCompound();
```

Comment:
You're not setting the NBT on the item anymore if it doesn't exist

Reply:
So what should I check? The `ItemStack `is being grabbed only when the packet runs so in case the book doesn't exist or for whatever reason in not in the hand slot anymore it won't get to this line of the code. 
The only thing this `if` is checking is if the book already had NBT or not since if it did it should only edit it and if it didn't it can be just set.

Reply:
That's not what I'm saying. You need to call book.setTagCompound if you're giving the item a completely new tag.

Reply:
Oh yes done